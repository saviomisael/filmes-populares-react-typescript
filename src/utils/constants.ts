export const IMAGE_URL = process.env.REACT_APP_IMAGE_URL;
