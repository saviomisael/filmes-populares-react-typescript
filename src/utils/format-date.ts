export const formatDate = (date: Date) =>
  date.toLocaleString('pt-BR', {
    year: 'numeric',
    month: 'long',
    day: 'numeric',
  });
