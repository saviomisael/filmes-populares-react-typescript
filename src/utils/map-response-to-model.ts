import { TvSerie } from 'models/TvSerie';
import { TvSerieResponse } from 'services/responses/TvSerieResponse';

const mapSingleResponseToTvSerieModel = (response: TvSerieResponse) =>
  new TvSerie(response.id, response.name, response.vote_average, response.poster_path);

export const mapResponseToTvSerieModel = (response: TvSerieResponse[]) =>
  response.map((x) => mapSingleResponseToTvSerieModel(x));
