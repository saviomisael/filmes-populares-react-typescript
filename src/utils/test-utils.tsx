import { render } from '@testing-library/react';
import { MemoryRouter, Switch, Route } from 'react-router-dom';
import { BaseStyles } from 'styles/base';
import { ThemeProvider } from 'styled-components';
import { theme } from 'styles/theme';
import { ReactNode } from 'react';
import { Provider } from 'react-redux';
import { storeMock } from 'mocks/configure-store-mock';

type optionsWithRouter = {
  initialRoutes?: string[];
  pathname: string;
};

export const renderTheme = (children: ReactNode) =>
  render(
    <ThemeProvider theme={theme}>
      <BaseStyles />
      {children}
    </ThemeProvider>,
  );

export const renderWithStoreProvider = (children: ReactNode) =>
  render(
    <Provider store={storeMock}>
      <ThemeProvider theme={theme}>
        <BaseStyles />
        {children}
      </ThemeProvider>
    </Provider>,
  );

export const renderWithRouter = (children: ReactNode, options: optionsWithRouter) => {
  const initialRoutes = options.initialRoutes && options.initialRoutes.length > 0 ? options.initialRoutes : ['/'];

  return render(
    <Provider store={storeMock}>
      <ThemeProvider theme={theme}>
        <MemoryRouter initialEntries={initialRoutes}>
          <Switch>
            <Route path={options.pathname} render={() => children} />
          </Switch>
        </MemoryRouter>
      </ThemeProvider>
    </Provider>,
  );
};
