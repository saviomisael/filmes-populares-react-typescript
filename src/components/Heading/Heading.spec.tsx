import { screen } from '@testing-library/react';
import { renderTheme } from 'utils/test-utils';
import { Heading } from '.';

describe('<Heading />', () => {
  it('should renders component with default props', () => {
    renderTheme(<Heading>Some text</Heading>);

    const heading = screen.getByRole('heading', { name: /some text/i });

    expect(heading.tagName.toLowerCase()).toBe('h1');

    expect(heading).toHaveStyleRule('color', '#FFFFFF');

    expect(heading).toHaveStyleRule('text-align', 'left');

    expect(heading).toHaveStyleRule('text-transform', 'none');
  });

  it('should renders component with text color dark when hasColorDark is true', () => {
    renderTheme(<Heading hasColorDark>Some text</Heading>);

    const heading = screen.getByRole('heading', { name: /some text/i });

    expect(heading).toHaveStyleRule('color', '#413B3B');
  });

  it('should renders as a h2 element', () => {
    renderTheme(<Heading as="h2">Some text</Heading>);

    const heading = screen.getByRole('heading', { name: /some text/i });

    expect(heading.tagName.toLowerCase()).toBe('h2');
  });

  it('should renders as a h3 element', () => {
    renderTheme(<Heading as="h3">Some text</Heading>);

    const heading = screen.getByRole('heading', { name: /some text/i });

    expect(heading.tagName.toLowerCase()).toBe('h3');
  });

  it('should renders as a h4 element', () => {
    renderTheme(<Heading as="h4">Some text</Heading>);

    const heading = screen.getByRole('heading', { name: /some text/i });

    expect(heading.tagName.toLowerCase()).toBe('h4');
  });

  it('should renders text align to center when hasTextCenter is true', () => {
    renderTheme(<Heading hasTextCenter>Some text</Heading>);

    const heading = screen.getByRole('heading', { name: /some text/i });

    expect(heading).toHaveStyleRule('text-align', 'center');
  });

  it('should renders text uppercase when hasTextUppercase is true', () => {
    renderTheme(<Heading hasTextUppercase>Some text</Heading>);

    const heading = screen.getByRole('heading', { name: /some text/i });

    expect(heading).toHaveStyleRule('text-transform', 'uppercase');
  });
});
