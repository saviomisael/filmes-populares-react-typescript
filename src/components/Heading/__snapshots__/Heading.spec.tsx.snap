// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should matches with snapshot 1`] = `
.c0 {
  color: #FFFFFF;
  text-align: left;
  text-transform: none;
  padding-bottom: 2rem;
}

<h1
  class="c0"
>
  Some text
</h1>
`;
